src/tests/functional/inference_engine/snippets/src/pass/insert_load_store.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+#include "pass/insert_load_store.hpp"
+#include "common_test_utils/common_utils.hpp"
+#include <subgraph_lowered.hpp>
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+std::string InsertLoadStoreTests::getTestCaseName(testing::TestParamInfo<multiInputParams> obj) {
+    std::vector<Shape> inputShapes(3);
+    std::vector<Shape> broadcastShapes(3);
+    std::tie(inputShapes[0], inputShapes[1], inputShapes[2],
+             broadcastShapes[0], broadcastShapes[1], broadcastShapes[2]) = obj.param;
+    std::ostringstream result;
+    for (size_t i = 0; i < inputShapes.size(); i++)
+        result << "IS[" << i << "]=" << CommonTestUtils::vec2str(inputShapes[i]) << "_";
+    for (size_t i = 0; i < broadcastShapes.size(); i++)
+        result << "BS[" << i << "]=" << CommonTestUtils::vec2str(broadcastShapes[i]) << "_";
+    return result.str();
+}
+
+void InsertLoadStoreTests::SetUp() {
+    TransformationTestsF::SetUp();
+    std::vector<Shape> inputShapes(3);
+    std::vector<Shape> broadcastShapes(3);
+    std::tie(inputShapes[0], inputShapes[1], inputShapes[2],
+             broadcastShapes[0], broadcastShapes[1], broadcastShapes[2]) = this->GetParam();
+    snippets_function = std::make_shared<EltwiseFunctionThreeInputsLowered>(inputShapes, broadcastShapes);
+}
+
+TEST_P(InsertLoadStoreTests, ThreeInputsEltwise) {
+    auto subgraph = getLoweredSubgraph(snippets_function->getOriginal());
+    function = subgraph->get_body();
+    function_ref = snippets_function->getLowered();
```

Comment:
Please, keep in mind that this is a lowering test, so the `function` should store a pointer to a lowered subgraph body. So we can initialize `function` in SetUp() only if we perform lowering there, but this is not an option because we actually testing the lowering.
The only thing we can do in regard of this comment is to initialize `function = snippets_function->getOriginal()`  in SetUp() and then reset it to `function = subgraph->get_body()` in TEST_P. However, I'm not sure it'll be clearer. What do you think?

Reply:
I am not sure about snippet functions and i need more details. But looks like what @eshoguli is right and this logic should be moved to `SetUp`

Reply:
Decided to leave it as is, since no better solutions were suggested.